compiler/control/CompilationController.hpp

```diff
@@ -22,24 +22,37 @@
 #ifndef COMPILATIONCONTROLLER_INCL
 #define COMPILATIONCONTROLLER_INCL
 
-#include "control/OptimizationPlan.hpp"
-#include "env/TRMemory.hpp"
+#include <stdint.h>
 
-class TR_MethodEvent;
-namespace TR { class Recompilation; }
+namespace TR { class CompilationStrategy; }
+namespace TR { class CompilationInfo; }
 
+//------------------------------- TR::CompilationController ------------------------
+// All methods and fields are static. The most important field is _compilationStrategy
+// that store the compilation strategy in use.
+//---------------------------------------------------------------------------------
 namespace TR
 {
-
-class DefaultCompilationStrategy : public TR::CompilationStrategy
+class CompilationController
    {
    public:
-   TR_PERSISTENT_ALLOC(TR_Memory::PersistentInfo);
-   DefaultCompilationStrategy() {}
-   TR_OptimizationPlan *processEvent(TR_MethodEvent *event, bool *newPlanCreated);
-   void shutdown() {}
-   virtual bool enableSwitchToProfiling() { return true; }
+   enum {LEVEL1=1, LEVEL2, LEVEL3}; // verbosity levels;
+   enum {DEFAULT, THRESHOLD}; // strategy codes
```

Comment:
I believe we can eliminate this enum since the threshold based strategy has been eliminated.

Reply:
Pls check [c2dbb06](https://github.com/eclipse/omr/commit/c2dbb061510edc21a29a4263dfd1bfdab0790477)

Reply:
Pls check [c2dbb06](https://github.com/eclipse/omr/commit/c2dbb061510edc21a29a4263dfd1bfdab0790477)